lib/app/routes/routes.dart

```diff
@@ -69,18 +72,27 @@ Route? onGenerateRoute({
                   );
                 case CreateInvoicePage.routeName:
                   return FadeInRoute(
-                    builder: (_) => const CreateInvoicePage(),
+                    builder: (_) => BlocProvider(
+                      create: (BuildContext context) => paymentLimitsCubit,
+                      child: const CreateInvoicePage(),
+                    ),
                     settings: settings,
                   );
                 case ReceiveChainSwapPage.routeName:
                   return FadeInRoute(
-                    builder: (_) => const ReceiveChainSwapPage(),
+                    builder: (_) => BlocProvider(
+                      create: (BuildContext context) => paymentLimitsCubit,
+                      child: const ReceiveChainSwapPage(),
+                    ),
                     settings: settings,
                   );
                 case SendChainSwapPage.routeName:
                   return FadeInRoute(
-                    builder: (_) => SendChainSwapPage(
-                      btcAddressData: settings.arguments as BitcoinAddressData?,
+                    builder: (_) => BlocProvider(
+                      create: (BuildContext context) => paymentLimitsCubit,
```

Comment:
If the cubit is instantiated globally then no reason to wrap it arround specific pages.
The idea was to create it per page so it will be "live" onlh during the page lifycycle and dispose it once the page exits.
I think either use it in the global blocs list or wrap on specific pages with create/dispose

Reply:
PaymentLimitsCubit is now created per page: [Fix PaymentLimitsCubit initialization](https://github.com/breez/l-breez/pull/63/commits/98ef3385a1015fe64fc5758c01e84b3b3d135260)